src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilder.java

```diff
@@ -286,12 +286,14 @@ public Pod build() {
                 	env.put("http_proxy", httpProxy);
                 }
             }
-        }
 
-        // Running on OpenShift Enterprise, security concerns force use of arbitrary user ID
-        // As a result, container is running without a home set for user, resulting into using `/` for some tools,
-        // and `?` for java build tools. So we force HOME to a safe location.
-        env.put("HOME", workingDir);
+            if (cloud.isOpenShift()) {
+                // Running on OpenShift Enterprise, security concerns force use of arbitrary user ID
+                // As a result, container is running without a home set for user, resulting into using `/` for some tools,
```

Comment:
Can we detect this condition directly before actually starting the container? If so, we would not need to have a different code path for OpenShift, and we might solve the problem for exactly the right set of use cases.